dpnp/fft/dpnp_utils_fft.py

```diff
@@ -423,10 +506,14 @@ def _validate_out_keyword(a, out, s, axes, c2r, r2c):
         # validate out shape against the final shape,
         # intermediate shapes may vary
         expected_shape = list(a.shape)
-        for s_i, axis in zip(s[::-1], axes[::-1]):
-            expected_shape[axis] = s_i
         if r2c:
-            expected_shape[axes[-1]] = expected_shape[axes[-1]] // 2 + 1
+            expected_shape[axes[-1]] = s[-1] // 2 + 1
+        if c2c:
```

Comment:
Otherwise it looks you are rewriting the same index in shape several times.
```suggestion
        elif c2c:
```